src/main/java/com/example/carservice/service/CarServiceImpl.java

```diff
@@ -10,13 +10,23 @@
 public class CarServiceImpl implements CarService {
 
     private final CarRepository repository;
+    private final CarRepository carRepository;
 
-    public CarServiceImpl(CarRepository repository) {
+    public CarServiceImpl(CarRepository repository, CarRepository carRepository) {
         this.repository = repository;
+        this.carRepository = carRepository;
     }
 
     @Override
     public List<Car> getCars() {
         return repository.findAll();
     }
+
+    @Override
+    public Car updateCar(){
+        Car carToUpdate= carRepository.findCarByModel("IS-250");
```

Comment:
Please make separate method in this class to find Car by model and call it here. 
Try to minimize number of request of repository. It's really important for performance.